File: src/main/java/nz/co/aetheric/sameness/Equatable.java
package nz.co.aetheric.sameness;

/**
 * Declares that something is isEquatable to another type of thing.
 * <p>Author: <a href="http://gplus.to/tzrlk">Peter Cummuskey</a></p>
 */
public interface Equatable<T> {

	boolean isEquatable(T right);

}

File: src/main/kotlin/nz/co/aetheric/sameness/Equatable.kt
package nz.co.aetheric.sameness

/**
 * Declares that something is isEquatable to another type of thing.
 *
 * Author: [Peter Cummuskey](http://gplus.to/tzrlk)
 */
interface Equatable<in T> {

	fun isEquatable(other: T?): Boolean

}
